### ISSUE TITLE
Transferable support

### ISSUE THREAD

**Issue by qm3ster** [developit/workerize-loader#51]
Hi[](https://nuxt.slack.com/messages/C3JC50TBP/)!
Do you think you could include a way to pass [`Transferable`](https://developer.mozilla.org/en-US/docs/Web/API/Transferable) objects correctly?
It would be great to be able to return a [`MessagePort`](https://developer.mozilla.org/en-US/docs/Web/API/MessagePort) so that the function can return a stream instead of a single value.

**Comment by developit** [comment]
Absolutely! We added it to greenlet in only a few bytes, the same solution should work reasonably well here.  It would only auto-detect transferables passed directly as arguments, not within nested objects:

https://github.com/developit/greenlet/blob/master/greenlet.js#L25-L29

**Comment by cpoopc** [comment]
look forward to it..

**Comment by NicolasRannou** [comment]
Hi I'm a bit late to the party but it seems unclear to me... Are transferable objects supported by `workerize` or not?

**Comment by developit** [comment]
They are not currently supported.

**Comment by shayke** [comment]
> Absolutely! We added it to greenlet in only a few bytes, the same solution should work reasonably well here. It would only auto-detect transferables passed directly as arguments, not within nested objects:
> 
> https://github.com/developit/greenlet/blob/master/greenlet.js#L25-L29

May I ask why only direct arguments? Why not detecting ArrayBuffers nested inside an object or an array?

**Comment by naoak** [comment]
I was thinking of a way to specify a loader option for how to extract the transferable from the arguments and return value of a worker method, but I realized that it would be impossible to determine the best option for each worker method with just one loader option. Instead, the approach I took was to call setup functions on both the worker and the main thread side to determine how to extract the transferables for each worker method. I've made a helper module to achieve this.

https://github.com/naoak/workerize-transferable
